WebAutopark.DAL/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebAutopark.DAL.Interfaces
+{
+    public interface IRepository<T> where T : class
+    {
+        Task Create(T instance);
+        Task Delete(int id);
+        Task<T> GetById(int id);
+        Task<IEnumerable<T>> GetAll();
+        Task Update(T instance);
+    }
```

Comment:
Back to sync variant
```suggestion
    public interface IRepository<T> where T : class
    {
        void Create(T instance);
        void Delete(int id);
        T GetById(int id);
        IEnumerable<T> GetAll();
        void Update(T instance);
    }
```

Reply:
Done